### ISSUE TITLE
Trap Server support needed with an easy to use API

### ISSUE THREAD

**Issue by sanchayanghosh** [lextudio/pysnmp#29]
PySNMP by etingof had a roundabout way of setting up a v2 TRAP server for receiving SNMP Traps and minimal support for V3 Trap.

We used pysnmp-bw in order to get v2 and v3 Trap support via SNMP, and the difference can be merged into pysnmp-lexstudio as well.

Would you be okay with me creating a PR and merging those changes (after writing a few UTs and checking the code?)

**Comment by lextm** [comment]
It is not a good idea to monitor TRAP/INFORM with Python due to performance limitations, but the changes in `pysnmp-bw` will be reviewed. If maintaining them isn't complicated, the commits will be cherry picked.

**Comment by lextm** [comment]
pysnmp-bw made a few changes in the underlying layers which currently we have no way to verify necessary or not. Thus, unable to move further on this item.

**Comment by sanchayanghosh** [comment]
Sure. Can take a look. I think there were a few PRs I had made there, seeing that there were a few bugs in asyncio at that point in time. I can take a look and see if it is necessary

**Comment by lextm** [comment]
After some further testing we found the changes broke all test cases developed so far. We decided to close this request,

1. The new types introduced in pysnmp-bw are not mandatory. We developed our own trap listener for unit testing and it wasn't much work, https://github.com/lextudio/pysnmp/blob/v6.0.6/tests/manager_context.py
2. In release 6.0 we fully switched to asyncio, and the patches in pysnmp-bw are not compatible at least for now.

We might revisit such an idea in the future.
